apps/studio/src/features/editing-experience/components/CreateCollectionModal/CreateCollectionModal.tsx

```diff
@@ -148,9 +156,25 @@ const CreateCollectionModalContent = ({
                   This will be applied to every child under this collection.
                 </FormHelperText>
               </FormLabel>
-              <Input
-                placeholder="This is a url for your new page"
-                {...register("permalink")}
+              <Controller
+                control={control}
+                name="permalink"
+                render={({ field: { onChange, ...field } }) => (
+                  <Input
+                    placeholder="This is a url for your new collection"
+                    noOfLines={1}
+                    maxLength={MAX_FOLDER_PERMALINK_LENGTH}
+                    {...field}
+                    onChange={(e) => {
+                      onChange(
```

Comment:
actually why do we need the slice? if users enter > than the allowed chars we just need to show validation error and prevent them from saving right

Reply:
i'm doing this to standardise validation across all our `Input`s. For example, our json forms renderer now also adopts this behaviour so it's just to 
1. have a consistsent behaviour across inputs in frontend
2. don't need to care about `disabled` because [illegal states are not representable](https://ybogomolov.me/making-illegal-states-unrepresentable)